### PR TITLE
ER-119, ER-121, ER-120

### PR DESCRIPTION
**Pull request by tranleduyan**
### Descriptions:
This PR includes:
1. ER - 119: Skeleton Layout Add To Inventory Page
    - Responsive Designs
3. ER - 121: Implement Equipment Model Addition API Calls
4. ER - 120: Implement Equipment Type Addition API Calls

It also checks for authorizations for navigations to the page. When testing and reviewing, please start both repos backend and frontend. If the ER-87-LDA-ModelAddition PR (PR#4) in the backend has not yet been merged, please run the branch instead of main branch.

### CODE REVIEW

File: src/Components/Buttons/FilterButton/FilterButton.css
Comment (mohammedking11-spu) [inline src/Components/Buttons/FilterButton/FilterButton.css:1]: Staying consistent with general styling, looks good!

File: src/Components/Buttons/FilterButton/FilterButton.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Import Components
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// Import Stylings
+import './FilterButton.css';
+
+// Import Icons
```
Comment (mohammedking11-spu) [inline src/Components/Buttons/FilterButton/FilterButton.js]: Import icons? Only a comment?

File: src/Components/Buttons/IconButton/IconButton.css
Comment (mohammedking11-spu) [inline src/Components/Buttons/IconButton/IconButton.css:1]: I appreciate the use of descriptive variable names 😀

File: src/Components/NavigationBar/NavigationBar.js
Comment (mohammedking11-spu) [inline src/Components/NavigationBar/NavigationBar.js:1]: I appreciate the consistency of the code.

File: src/Components/Buttons/FilterButton/FilterButton.js
Comment (mohammedking11-spu) [inline src/Components/Buttons/FilterButton/FilterButton.js:1]: This file looks good, but you might have missed importing the icons.

**Review by mohammedking11-spu — APPROVED** [review]
ER-119-121-120 all looks good. I reviewed them and the code works and is written in a very clean and sustainable way. The code also fulfills the tasks, and the comments made the code much easier to understand, even with lots of files to keep up with, it made it easier. I will approve it on my end, and just make sure to review my comment about the import icons. Thanks Le!

File: src/Components/Buttons/FilterButton/FilterButton.js (hunk 1 above)
Comment (tranleduyan) [reply src/Components/Buttons/FilterButton/FilterButton.js]: Thank you! I just removed this unnecessary comments!

**Review by amansiid — COMMENTED** [review]
Styling looks good and headers tell the brief  purposes.

File: src/Pages/AddToInventoryPage/AddToInventoryPage.js
Comment (amansiid) [inline src/Pages/AddToInventoryPage/AddToInventoryPage.js:1]: Very good inline comments it helps to follow along

File: src/Pages/DashboardPage/DashboardPage.js
Comment (amansiid) [inline src/Pages/DashboardPage/DashboardPage.js:1]: Dashboard page looks good to me

**Review by amansiid — APPROVED** [review]
Reviewed code and tested the APIs everything looks functional. 
Good Job!

**Pull request merged**